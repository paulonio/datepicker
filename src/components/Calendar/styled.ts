import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
  padding: 10px 10px 0;
  max-width: 252px;
  border: solid 1px #e1e1e1;
  border-radius: 8px;
`;

export const WeekWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #333333;
  font-size: 14px;
  font-weight: 700;
  line-height: 19px;
`;

export const ClearButton = styled.button`
  width: 100%;
  padding: 10px 0;
  display: flex;
  justify-content: center;
  border: none;
  border-top: solid 1px #e1e1e1;
  color: #333333;
  background-color: #ffffff;
  font-size: 12px;
  font-weight: 600;
  line-height: 16px;
  &:hover {
    cursor: pointer;
  }
`;
