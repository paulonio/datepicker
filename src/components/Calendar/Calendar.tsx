import React, { FC } from 'react';
import WeekDayRow from '../WeekDayRow/WeekDayRow';
import MonthRow from '../MonthRow/MonthRow';
import { ClearButton, Wrapper } from './styled';
import DateButtons from '../DateButtons/DateButtons';
import { usePanelDates } from '../../hooks/usePanelDates';
import { useDisplayDates } from '../../hooks/useDisplayDates';

export interface CalendarProps {
  fromTo: 'From' | 'To';
  dateFrom: Date;
  dateTo: Date;
  onDateFromChange: (date: Date) => void;
  onDateToChange: (date: Date) => void;
}

const Calendar: FC<CalendarProps> = ({
  fromTo,
  dateFrom,
  dateTo,
  onDateFromChange,
  onDateToChange,
}) => {
  const { month, year, setMonth, setYear } = usePanelDates(fromTo, dateFrom, dateTo);
  const allDays = useDisplayDates(month, year);

  return (
    <Wrapper>
      <MonthRow month={month} year={year} handleMonthChange={setMonth} handleYearChange={setYear} />
      <WeekDayRow />
      <DateButtons
        dateFrom={dateFrom}
        dateTo={dateTo}
        allDays={allDays}
        onDateChange={fromTo === 'From' ? onDateFromChange : onDateToChange}
      />
      <ClearButton>Clear</ClearButton>
    </Wrapper>
  );
};

export default Calendar;
