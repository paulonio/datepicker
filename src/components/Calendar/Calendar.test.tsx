import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import type { DatepickerProps } from '@components/Datepicker/Datepicker';
import Calendar from './Calendar';
import type { Init } from '@/types/types';

describe('Calendar', () => {
  const config: DatepickerProps = {
    start: 'mo',
    view: 'monthly',
    weekend: 'show',
    mode: 'range',
    minDate: new Date(2022, 4, 11),
    maxDate: new Date(2024, 4, 11),
  };

  const state: Init = {
    currentCalendar: 'From',
    showCalendar: true,
    date: new Date(2023, 4, 16),
    from: new Date(2023, 4, 2),
    to: new Date(2023, 4, 30),
  };

  it('should render correctly calendar', () => {
    render(<Calendar config={config} state={state} dispatch={() => {}} />);

    const title = screen.getByTestId('month-row');
    expect(title).toHaveTextContent('May 2023');
  });
});
