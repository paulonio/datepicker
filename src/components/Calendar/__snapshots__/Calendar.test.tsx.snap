// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should render calendar 1`] = `
<div>
  <div
    class="sc-hAtEyd fvCzsn"
  >
    <div
      class="sc-fmSAUk jXhzZk"
      data-component="calendar"
      data-testid="calendar"
    >
      <label
        class="sc-blLsxD bVvqgW"
      >
        <input
          class="sc-ieZDjg"
          type="checkbox"
        />
        Select one date
      </label>
      <div
        class="sc-hLseeU kGWQkR"
      >
        <img
          class="sc-gLDzan hCdWEK"
          data-testid="prev-year"
          src="./assets/icons/Prev-year.svg"
        />
        <img
          class="sc-eDDNvR cBNGiP"
          data-testid="prev"
          src="./assets/icons/Prev.svg"
        />
        <div
          class="sc-dmqHEX lbuZOD"
          data-testid="month-row"
        >
          May
           
          2023
        </div>
        <img
          class="sc-jTrPJq hLJulf"
          data-testid="next"
          src="./assets/icons/Next.svg"
        />
        <img
          class="sc-iAEyYk hCGgmq"
          data-testid="next-year"
          src="./assets/icons/Next-year.svg"
        />
      </div>
      <div
        class="sc-beqWaB fleLma"
      >
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          Mo
        </div>
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          Tu
        </div>
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          We
        </div>
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          Th
        </div>
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          Fr
        </div>
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          Sa
        </div>
        <div
          class="sc-gueYoa fUeWBE"
          data-testid="week-day"
        >
          Su
        </div>
      </div>
      <div
        class="sc-fsQiph fjgByX"
      >
        <button
          class="sc-qRumB hiswWa"
          data-testid="date-button"
        >
          1
        </button>
        <button
          class="sc-qRumB gwncVz"
          data-testid="date-button"
        >
          2
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          3
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          4
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          5
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          6
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          7
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          8
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          9
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          10
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          11
        </button>
        <button
          class="sc-qRumB eZcKpY"
          data-testid="date-button"
        >
          12
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          13
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          14
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          15
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          16
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          17
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          18
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          19
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          20
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          21
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          22
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          23
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          24
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          25
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          26
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          27
        </button>
        <button
          class="sc-qRumB gaVQPJ"
          data-testid="date-button"
        >
          28
        </button>
        <button
          class="sc-qRumB kOJDLz"
          data-testid="date-button"
        >
          29
        </button>
        <button
          class="sc-qRumB ictflA"
          data-testid="date-button"
        >
          30
        </button>
        <button
          class="sc-qRumB hiswWa"
          data-testid="date-button"
        >
          31
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          1
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          2
        </button>
        <button
          class="sc-qRumB gaghwX"
          data-testid="date-button"
        >
          3
        </button>
        <button
          class="sc-qRumB gaghwX"
          data-testid="date-button"
        >
          4
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          5
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          6
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          7
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          8
        </button>
        <button
          class="sc-qRumB fipidh"
          data-testid="date-button"
        >
          9
        </button>
        <button
          class="sc-qRumB gaghwX"
          data-testid="date-button"
        >
          10
        </button>
        <button
          class="sc-qRumB gaghwX"
          data-testid="date-button"
        >
          11
        </button>
      </div>
      <button
        class="sc-dKfzgJ kjZgqV"
      >
        Clear
      </button>
    </div>
  </div>
</div>
`;
